File: easy/3875-construct-uniform-parity-array-i/solution.java
/**
 * 3875. Construct Uniform Parity Array I
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public boolean canConstruct(int[] nums1) {
        int odd = 0;
        for (int x : nums1) {
            if (x % 2 != 0) {
                odd++;
            }
        }
        int even = nums1.length - odd;

        // An odd element must flip parity, needing another odd at a different index.
        boolean allEven = odd != 1;
        // An even element must flip parity, needing any odd element.
        boolean allOdd = even == 0 || odd >= 1;

        return allEven || allOdd;
    }
}

File: easy/3875-construct-uniform-parity-array-i/solution.py
from typing import List


class Solution:
    def canConstruct(self, nums1: List[int]) -> bool:
        """
        3875. Construct Uniform Parity Array I
        Time: O(n)
        Space: O(1)
        """
        odd = sum(1 for x in nums1 if x % 2)
        even = len(nums1) - odd

        # An odd element must flip parity, needing another odd at a different index.
        all_even = odd != 1
        # An even element must flip parity, needing any odd element.
        all_odd = even == 0 or odd >= 1

        return all_even or all_odd
